python/ray/data/preprocessors/vectorizer.py

```diff
@@ -248,15 +249,17 @@ def most_common(counter: Counter, n: int):
 
     def _transform_pandas(self, df: pd.DataFrame):
 
+        to_concat = []
         for col in self.columns:
             token_counts = self.stats_[f"token_counts({col})"]
             sorted_tokens = [token for (token, count) in token_counts.most_common()]
             tokenized = df[col].map(self.tokenization_fn).map(Counter)
             for token in sorted_tokens:
-                df[f"{col}_{token}"] = tokenized.map(lambda val: val[token])
+                series = tokenized.map(lambda val: val[token])
+                series.name = f"{col}_{token}"
+                to_concat.append(series)
 
-        # Drop original columns.
-        df.drop(columns=self.columns, inplace=True)
+        df = pd.concat(to_concat, axis=1)
```

Comment:
Is this change related to the bug?

Reply:
This is a more performant way of constructing the dataframe